bodhi/server/__init__.py

```diff
@@ -45,23 +44,28 @@
 
 def get_db_session_for_request(request=None):
     """
-    This function returns a database session that is meant to be used for the given request. It sets
-    up the Zope transaction manager and configures the request to close the session when it is
-    completed. If you need a database session that is not tied to a request, you can use
-    bodhi.server.models.get_db_factory() to return a session generator.
+    This function returns a database session that is meant to be used for the given request.
+
+    It handles rolling back or committing the session based on whether an exception occurred or
+    not. To get a database session that's not tied to the request/response cycle, just use the
+    :data:`Session` scoped session in this module.
+
+    Args:
+        request (pyramid.request): The request object to create a session for.
+
+    Returns:
+        sqlalchemy.orm.session.Session: A database session.
     """
-    engine = engine_from_config(request.registry.settings, 'sqlalchemy.')
-    Sess = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
-    Sess.configure(bind=engine)
-    session = Sess()
+    session = request.registry.sessionmaker()
 
     def cleanup(request):
-        # No need to do rollback/commit ourselves.  the zope transaction manager takes care of that
-        # for us. However, we want to explicitly close the session we opened
+        if request.exception is not None:
```

Comment:
Though this function didn't actually have a docblock before, the comment that has been removed here was kind of like one. So it would be good to give it a docblock describing the new behavior.